ffi/athcon/bindings/rust/athcon-vm/src/encode_tx.rs

```diff
@@ -0,0 +1,137 @@
+//! Implements athcon_encode_tx
+use athcon_sys as ffi;
+use athena_interface::transaction::{Encode, Transaction};
+use athena_vm_sdk::{Pubkey, SendArguments};
+
+/// Encode Athena Transaction into bytes.
+///
+/// # Arguments
+/// - `principal` - The transaction's principal account.
+/// - `template` - Optional transaction's template account.
+/// - (`method`, `method_size`) - Optional method to call
+/// - `nonce` - The transaction's nonce
+/// - (`args`, `args_size`) - Optional transaction's serialized arguments.
+///
+/// # Safety
+/// - the caller is responsible for freeing the returned vector
+///   via the `athcon_free_vector` function.
+#[no_mangle]
+unsafe extern "C" fn athcon_encode_tx(
+  principal: *const ffi::athcon_address,
+  template: *const ffi::athcon_address,
+  method: *const u8,
+  method_size: usize,
+  nonce: u64,
+  args: *const u8,
+  args_size: usize,
+) -> *mut ffi::athcon_bytes {
+  let principal = &(*principal).bytes;
+  let template_o = template.as_ref().map(|t| t.bytes);
+  let method = if method.is_null() {
+    &[]
+  } else {
+    std::slice::from_raw_parts(method, method_size)
+  };
+  let args = if args.is_null() {
+    &[]
+  } else {
+    std::slice::from_raw_parts(args, args_size)
+  };
+
+  let tx = Transaction::new(nonce, *principal, template_o, method, args);
+
+  let (ptr, size) = crate::allocate_output_data(tx.encode());
+
+  Box::into_raw(Box::new(ffi::athcon_bytes { ptr, size }))
+}
+
+#[no_mangle]
+unsafe extern "C" fn athcon_encode_tx_spawn(
+  pubkey: *const ffi::athcon_bytes32,
+) -> *mut ffi::athcon_bytes {
+  let encoded = Pubkey((*pubkey).bytes).encode();
+  let (ptr, size) = crate::allocate_output_data(encoded);
+
+  Box::into_raw(Box::new(ffi::athcon_bytes { ptr, size }))
+}
+
+#[no_mangle]
+unsafe extern "C" fn athcon_encode_tx_send(
+  recipient: *const ffi::athcon_address,
+  amount: u64,
+) -> *mut ffi::athcon_bytes {
+  let args = SendArguments {
```

Comment:
should probably rename this struct `SpendArguments` for consistency with go-spacemesh:

https://github.com/spacemeshos/go-spacemesh/blob/5a40816d2bd503f53c9fa5a0476fe7fd2d8506db/genvm/templates/wallet/types.go#L14-L18